### PR TITLE
Feature/query cxx iasi

### PR DESCRIPTION
**Pull request by emilyhcliu**
## Description

*(Instructions: replace text in this and all sections with your own text)*

Provide a detailed description of what this PR does.

What problem does it fix? What new capability does it add?

[See the JEDI Documentation](https://jointcenterforsatellitedataassimilation-jedi-docs.readthedocs-hosted.com/en/latest/inside/practices/pullrequest.html) for further information.

### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #<issue_number>

If this does not close an existing issue, then [be sure to add a Pipeline, Label, Estimate, Assignees, and Epic](https://jointcenterforsatellitedataassimilation-jedi-docs.readthedocs-hosted.com/en/latest/inside/practices/issues.html)

## Acceptance Criteria (Definition of Done)

What does it mean for this PR to be finished?

### CODE REVIEW

File: test/testinput/bufr_ncep_mtiasi.yaml [hunk 1]
```diff
@@ -197,7 +194,13 @@ observations:
...
 
         # ObsValue
-        - name: "ObsValue/scaledRadiance"
-          source: variables/scaledRadiance
-          longName: "Scaled IASI Radiance"
+        - name: "ObsValue/scaledSpectralRadiance"
+          source: variables/scaledSpectralRadiance
+          longName: "Scaled IASI Spectral Radiance"
           units: "W m-2 sr-1 m"
+
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_mtiasi.yaml:201]: units: "W m-2 sr-1 m", I checked unit is "units: "W m-2 sr-1" without m. Could you please double check?

File: test/testinput/bufr_ncep_mtiasi.yaml [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+          longitude:
+            query: "*/CLONH"
+
+          satelliteId:
+            query: "*/SAID"
+
+          sensorId:
+            query: "*/SIID[1]"
+
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_mtiasi.yaml:33]: query: "*/SIID[1]" or query: "*/SIID" ??

File: test/testinput/bufr_ncep_mtiasi.yaml [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+          longName: "Sensor Azimuth Angle"
+          units: "degree"
+          range: [ 0, 360 ]
+
+        - name: "MetaData/heightOfStation"
+          source: variables/heightOfStation
+          longName: "Altitude of Satellite"
+          units: "m"
+          range: [ 0, 360 ]
+
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_mtiasi.yaml:179]: range: [ 0, 360 ] for heightOfStation ??

File: test/testinput/bufr_ncep_mtiasi.yaml [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+          source: variables/sensorId
+          longName: "Satellite Instrument"
+
+        - name: "MetaData/scanLineNumber"
+          source: variables/scanLineNumber
+          longName: "Scan Line Number"
+
+        - name: "MetaData/qualityFlags"
+          source: variables/gqisFlagQual
+          longName: "Individual IASI-System Quality Flag"
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_mtiasi.yaml:144]: source: variables/gqisFlagQual => source: variables/qualityFlags to keep the consistency.

File: test/testinput/bufr_ncep_mtiasi.yaml [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+          sensorId:
+            query: "*/SIID[1]"
+
+          scanLineNumber:
+            query: "*/SLNM"
+
+          gqisFlagQual:
+            query: "*/QGFQ"
+
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_mtiasi.yaml:39]: gqisFlagQual: => qualityFlags: ??

File: test/testinput/bufr_ncep_mtiasi.yaml [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+        - name: "MetaData/dateTime"
+          source: variables/timestamp
+          longName: "Datetime"
+          units: "seconds since 1970-01-01T00:00:00Z"
+
+        - name: "MetaData/latitude"
+          source: variables/latitude
+          longName: "Latitude"
+          units: "degrees_north"
+          range: [ -90, 90 ]
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_mtiasi.yaml:121]: units: "degrees_north" or units: "degree_north" ??

File: test/testinput/bufr_ncep_mtiasi.yaml [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+        - name: "MetaData/longitude"
+          source: variables/longitude
+          longName: "Longitude"
+          type: float 
+          units: "degrees_east"
+          range: [ -180, 180 ]
+
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_mtiasi.yaml:129]: The same above

File: test/testinput/bufr_ncep_mtiasi.yaml [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+        - name: "MetaData/satelliteIdentifier"
+          source: variables/satelliteId
+          longName: "Satellite Identifier"
+
+        - name: "MetaData/satelliteInstrument"
+          source: variables/sensorId
+          longName: "Satellite Instrument"
+
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_mtiasi.yaml:137]: instrumentIdentifier, see convention table.  Double check if you need to correct ??

File: test/testinput/bufr_ncep_mtiasi.yaml [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+        - name: "MetaData/longitude"
+          source: variables/longitude
+          longName: "Longitude"
+          type: float 
+          units: "degrees_east"
+          range: [ -180, 180 ]
+
+        - name: "MetaData/satelliteIdentifier"
+          source: variables/satelliteId
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_mtiasi.yaml:131]: source: variables/satelliteId => source: variables/satelliteIdentifier

File: test/testinput/bufr_ncep_mtiasi.yaml [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+              hour: "*/HOUR"
+              minute: "*/MINU"
+
+          latitude:
+            query: "*/CLATH"
+
+          longitude:
+            query: "*/CLONH"
+
+          satelliteId:
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_mtiasi.yaml:28]: satelliteId: =>  satelliteIdentifier:

File: test/testinput/bufr_ncep_mtiasi.yaml [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+            query: "*/SAID"
+
+          sensorId:
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_mtiasi.yaml:31]: sensorId: => instrumentIdentifier

File: test/testinput/bufr_ncep_mtiasi.yaml [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+          longName: "Longitude"
+          type: float 
+          units: "degrees_east"
+          range: [ -180, 180 ]
+
+        - name: "MetaData/satelliteIdentifier"
+          source: variables/satelliteId
+          longName: "Satellite Identifier"
+
+        - name: "MetaData/satelliteInstrument"
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_mtiasi.yaml:134]: satelliteInstrument:  => instrumentIdentifier:

File: test/testinput/bufr_ncep_mtiasi.yaml [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+          source: variables/sensorId
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_mtiasi.yaml:135]: source: variables/sensorId => source: variables/instrumentIdentifier

File: test/testinput/bufr_ncep_mtiasi.yaml [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+          longName: "Satellite Instrument"
```
Comment (YoulongXia-NOAA) [inline test/testinput/bufr_ncep_mtiasi.yaml:136]: longName: "Satellite Instrument" => longName: "Satellite Instrument Identifier"

**Comment by emilyhcliu** [comment]
@YoulongXia-NOAA @nicholasesposito I addressed all the questions and suggestions.  Thanks for your reviews.

**Review by nicholasesposito — APPROVED** [review]

**Review by YoulongXia-NOAA — APPROVED** [review]
My comments have been addressed. I approve this PR.

**Comment by emilyhcliu** [comment]
> My comments have been addressed. I approve this PR.

Yes. Nick had fixed the file extension for prepbufr file (...adpsfc.prepbufr).
We are good to merge this PR!

**Pull request merged**